File: src/plugin/combat/special/melee/ArmadylGodswordSpecialAttackPlugin.java
package plugin.combat.special.melee;

import org.redrune.game.content.combat.player.AbstractCombatStyle;
import org.redrune.game.entity.actor.Actor;
import org.redrune.game.entity.actor.mask.Animation;
import org.redrune.game.entity.actor.mask.Graphics;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.plugin.combat.SpecialAttackPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/5/2017
 */
public class ArmadylGodswordSpecialAttackPlugin extends SpecialAttackPlugin {
	
	private static final Animation ANIMATION = new Animation(11989);
	
	private static final Graphics GRAPHICS = new Graphics(2113);
	
	@Override
	public int[] getWeaponIds() {
		return arguments(11694);
	}
	
	@Override
	public void fire(Player source, Actor target, AbstractCombatStyle style) {
		source.setNextAnimation(ANIMATION);
		source.setNextGraphics(GRAPHICS);
		style.sendHit(source, target, style.getCalculator().getMaximumHit(source,1.375), style.getRandomDamage(source, target, 1.375), 0);
	}
	
}

File: src/main/java/plugin/interaction/combat/special/melee/ArmadylGodswordSpecialAttackPlugin.kt
package plugin.interaction.combat.special.melee

import game.content.entity.actor.combat.player.AbstractCombatStyle
import game.content.plugin.combat.SpecialAttackPlugin
import game.entity.actor.Actor
import game.entity.actor.mask.Animation
import game.entity.actor.mask.Graphics
import game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 9/5/2017
 */
class ArmadylGodswordSpecialAttackPlugin : SpecialAttackPlugin() {
    override fun getWeaponIds(): IntArray {
        return arguments(11694)
    }

    override fun fire(source: Player, target: Actor, style: AbstractCombatStyle) {
        source.nextAnimation = ANIMATION
        source.setNextGraphics(GRAPHICS)
        style.sendHit(
            source,
            target,
            style.calculator.getMaximumHit(source, 1.375),
            style.getRandomDamage(source, target, 1.375),
            0
        )
    }

    companion object {
        private val ANIMATION = Animation(11989)
        private val GRAPHICS = Graphics(2113)
    }
}